### PR TITLE
more User Guide updates & fixes

### PR DESCRIPTION
**Pull request by preminger**
## Description of the Pull Request (PR):

This is a continuation of the work begun in https://github.com/sylabs/singularity-userdocs/pull/161 – going through the various parts of the User Guide, improving clarity, finding & fixing potential problems, etc.

### CODE REVIEW

File: definition_files.rst [hunk 1]
```diff
@@ -480,56 +518,65 @@ hashbang (``#!``) as the first line in your ``%test`` section:
...
+*container*, and since ``zsh`` is not built into the base Ubuntu image,
+it would have to be installed as part of the ``%post`` section for this
+``%test`` code to work properly.
 
 %environment
 ============
 
+.. note::
+
+   The sections detailed from this point forward are all applied to the
```
Comment (dtrudg) [inline definition_files.rst]: I'm not entirely clear what is meant by this.

**Review by dtrudg — APPROVED** [review]
LGTM so far. Thanks!

File: definition_files.rst (hunk 1 above)
Comment (preminger) [reply definition_files.rst]: This was a slight modification of text that was already there before. The idea, as I understand it, was that unlike, e.g., the `%setup`, `%post`, `%runscript` sections discussed until that point in the doc, which are intrinsically ordered & discrete in how they are applied to the container being built, no such ordering can be assumed by the user for the sections discussed from this point forward in the Definition Files document.

Having said that, if that's either incorrect, or it is correct but there's a better way of saying it (or maybe it's correct but still not worth saying!), let me know.

Comment (dtrudg) [reply definition_files.rst]: I would say we drop it. There is an order in which they are inserted to the container. Also, there is ordering in how `%environment` is handled at runtime ahead of e.g. `%runscript`.

I think it's somewhat confusing, and not really needed.

File: build_env.rst [hunk 2]
```diff
@@ -24,13 +24,13 @@ Cache Folders
 {Singularity} will cache SIF container images generated from remote
 sources, and any OCI/docker layers used to create them. The cache is
 created at ``$HOME/.singularity/cache`` by default. The location of the
-cache can be changed by setting the ``SINGULARITY_CACHEDIR`` environment
+cache can be changed using the ``SINGULARITY_CACHEDIR`` environment
```
Comment (dtrudg) [inline build_env.rst]: Personally prefer 'setting' here. If someone searches for 'setting environment variable' then it's more likely to give them what they need to know.

**Review by dtrudg — APPROVED** [review]
Minor comments only. Thanks!

**Pull request merged**